File: leetCode/problems/medium/leetcode_53/leetcode_53_maxSubArray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        currentMax = nums[0]
        currentSum = 0
        for num in nums:
            if currentSum < 0:
                currentSum = 0
            currentSum += num
            currentMax = max(currentMax, currentSum)
        return currentMax

File: leetCode/problems/medium/leetcode_53/leetcode_53_maxSubArray.js
var maxSubArray = function(nums) {
    currentMax = nums[0]
    currentSum = 0
    for(let n of nums){
        if(currentSum < 0){
            currentSum = 0
        };
        currentSum += n
        currentMax = Math.max(currentMax, currentSum)
    };
    return currentMax;
};